File: summoner.cxx
#include "summoner.h"
#include "leagueEnum.h"
#include "string"

Summoner::Summoner():
sname("Default"), 
pname("Default"), 
league(LeagueEnums::unranked), 
division(1), 
primaryRole(LeagueEnums::noroll),
secndryRole(LeagueEnums::noroll)
{}

void Summoner::setSummonerName(const std::string& name){
    sname = name;
}

void Summoner::setPlayerName(const std::string& name){
    pname = name;
}

void Summoner::setLeague(const LeagueEnums::League newLeague){
    league = newLeague;
}

void Summoner::setDivision(const int div){
    division = div;
}

void Summoner::setPrimaryRole(LeagueEnums::Role prim){
    primaryRole = prim;
}

void Summoner::setSecndryRole(LeagueEnums::Role scnd){
    secndryRole = scnd;
}

std::string Summoner::getSummonerName() const{
    return sname;
}

std::string Summoner::getPlayerName() const{
    return pname;
}

LeagueEnums::League Summoner::getLeague() const{
    return league;
}

int Summoner::getDivision() const{
    return division;
}

int Summoner::getRank() const{
    if (league == LeagueEnums::unranked or 
        league == LeagueEnums::masters or
        league == LeagueEnums::challenger)
    {
        return league;
    }
    return league*5 - (5 - division);
}

LeagueEnums::Role Summoner::getPrimaryRole() const{
    return primaryRole;
}

LeagueEnums::Role Summoner::getSecndryRole() const{
    return secndryRole;
}

File: league_gen_py/summoner.py
__author__ = 'ryan'
# TODO:
# make league and division automatic when summoner name or ID is supplied.


class Summoner:
    def __init__(self):
        self.summonerName = ""
        self.summonerID = ""
        self.league = ""
        self.division = ""
        self.primaryRole = ""
        self.secondaryRole = ""

    def set_summoner_name(self, name):
        self.summonerName = name

    def set_summoner_id(self):
        self.summonerID = ""

    def set_league(self, league):
        self.league = league

    def set_division(self, division):
        self.division = division

    def set_primary_role(self, primaryRole):
        self.primaryRole = primaryRole

    def set_secondary_role(self, secondaryRole):
        self.secondaryRole = secondaryRole

    def get_summoner_name(self):
        return self.summonerName

    def get_summoner_id(self):
        return self.summonerID

    def get_league(self):
        return self.league

    def get_division(self):
        return self.division

    def get_primary_role(self):
        return self.primaryRole

    def get_secondary_role(self):
        return self.secondaryRole
